acceptance: TestDockerCLI_test_sql_mem_monitor failed

acceptance.TestDockerCLI_test_sql_mem_monitor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/14772047?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/14772047?buildTab=artifacts#/) on master @ [911f1ce389342459592d89fe7b78bcba3a2d265a](https://github.com/cockroachdb/cockroach/commits/911f1ce389342459592d89fe7b78bcba3a2d265a):


```
=== RUN   TestDockerCLI_test_sql_mem_monitor
    test_log_scope.go:170: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/534f11b5b7b50f31261cf1b2945268b5/logTestDockerCLI_test_sql_mem_monitor241083552
    test_log_scope.go:81: use -show-logs to present logs inline
    generated_cli_test.go:425: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/534f11b5b7b50f31261cf1b2945268b5/logTestDockerCLI_test_sql_mem_monitor241083552
--- FAIL: TestDockerCLI_test_sql_mem_monitor (69.99s)
=== RUN   TestDockerCLI_test_sql_mem_monitor/test_sql_mem_monitor.tcl
    util_docker.go:259: non-zero exit code: 1
    --- FAIL: TestDockerCLI_test_sql_mem_monitor/test_sql_mem_monitor.tcl (64.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_sql_mem_monitor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37687

--- a/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
+++ b/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
@@ -79,12 +79,13 @@ set spawn_id $shell_spawn_id
 # Error is either an explicit "out of memory" emitted by Go or one of the errors
 # thrown by CGo.
 #
-# "fatal error: unexpected signal during runtime execution" might seem like it
-# doesn't belong here, but it's an artifact of how we're limiting the memory
-# usage of the CRDB process. In particular, ulimit is a "user limit" which is
-# enforced not at the OS kernel level (i.e. not via oomkiller) but at the
-# process execution level. As a result, memory allocation error doesn't kill the
-# process, so it keeps on running and can hit this "fatal error" later on.
+# "fatal error: unexpected signal during runtime execution" and "segmentation
+# violation" might seem like they don't belong here, but it's an artifact of how
+# we're limiting the memory usage of the CRDB process. In particular, ulimit is
+# a "user limit" which is enforced not at the OS kernel level (i.e. not via
+# oomkiller) but at the process execution level. As a result, memory allocation
+# error doesn't kill the process, so it keeps on running and can hit this "fatal
+# error" or "segmentation violation" later on.
 expect {
     "out of memory" {}
     "cannot allocate memory" {}
@@ -92,6 +93,7 @@ expect {
     "Resource temporarily unavailable" {}
     "_Cfunc_calloc" {}
     "fatal error: unexpected signal during runtime execution" {}
+    "segmentation violation" {}
     timeout { handle_timeout "memory allocation error" }
 }
 # Stop the tail command.

